src/error.rs

```diff
@@ -16,4 +16,7 @@ pub enum Error {
 
     /// The input doesn't yield enough input elements.
     TooFewInputElements,
+
+    /// Failed to decrypt the message with the provided secret and nonce
+    IncorrectEncryption,
```

Comment:
Isn't this more like `DecryptionFailed`?

Reply:
Totally. It's a chicken - egg problem: when the decryption failed, the encryption wasn't done correctly either. But I also like `DecryptionFailed`

Reply:
True. I find the first order - i.e. from the user's perspective - to be that the decryption failed, while from the implementer's perspective it would indeed be that the encryption was incorrect.